src/lib.rs

```diff
@@ -602,6 +644,28 @@ impl Consul {
     }
 }
 
+fn add_query_option_params(uri: &mut String, query_opts: &QueryOptions, mut separator: char) {
```

Comment:
Is there not a type in hyper/std that allows us to build this URL w/o doing string manipulation?

Reply:
there is `http::Uri` but that requires passing the path and query params as either string or bytes